sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.institutions.controller.searchStudents.e2e-spec.ts

```diff
@@ -0,0 +1,402 @@
+      firstName: "",
+      lastName: "",
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(collegeFInstitutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect([
+        {
+          id: student.id,
+          firstName: student.user.firstName,
+          lastName: student.user.lastName,
+          birthDate: student.birthDate,
+          sin: student.sinValidation.sin,
+        },
+      ]);
+  });
+
+  it("Should not find the student when user is not active.", async () => {
+    // Arrange
+    const { student, application } =
+      await saveStudentWithApplicationForCollegeF(ApplicationStatus.Submitted);
+    student.user.isActive = false;
+    await studentRepo.save(student);
+
+    const searchPayload = {
+      appNumber: application.applicationNumber,
+      firstName: "",
+      lastName: "",
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(collegeFInstitutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect([]);
+  });
+
+  it("Should not find the student when student does not have an application for that institution.", async () => {
+    // Arrange
+    const { application } = await saveStudentWithApplicationForCollegeF(
+      ApplicationStatus.Submitted,
+    );
+
+    const { institution: collegeC } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    const collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    const collegeCInstitutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    const searchPayload = {
+      appNumber: application.applicationNumber,
+      firstName: "",
+      lastName: "",
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(collegeCInstitutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect([]);
+  });
+
+  it("Should not find the student when student has only a draft application.", async () => {
+    // Arrange
+    const { application } = await saveStudentWithApplicationForCollegeF(
+      ApplicationStatus.Draft,
+    );
+    const searchPayload = {
+      appNumber: application.applicationNumber,
+      firstName: "",
+      lastName: "",
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(collegeFInstitutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect([]);
+  });
+
+  const saveStudentWithApplicationForCollegeF = async (
```

Comment:
Since this is a new method, hence it would need comments and a return type (which can be an anonymous type to keep it simple), unless there is a team agreement to skip those.
@ann-aot @dheepak-aot @guru-aot @sh16011993